Lib/test/test_asyncio/test_waitfor.py

```diff
@@ -277,27 +279,77 @@ async def return_42():
 
 
     async def test_wait_for_issue86296(self):
-        # See https://github.com/python/cpython/issues/86296
+        # GH-86296: The task should get cancelled and not run to completion.
+        # inner completes in one cycle of the event loop so it
+        # completes before the task is cancelled.
 
         async def inner():
-            return
+            return 'done'
 
+        inner_task = asyncio.create_task(inner())
         reached_end = False
 
         async def wait_for_coro():
-            await asyncio.wait_for(inner(), timeout=100)
+            await asyncio.wait_for(inner_task, timeout=100)
             await asyncio.sleep(1)
             nonlocal reached_end
             reached_end = True
 
         task = asyncio.create_task(wait_for_coro())
-        await asyncio.sleep(0)
         self.assertFalse(task.done())
+        # Run the task
+        await asyncio.sleep(0)
         task.cancel()
         with self.assertRaises(asyncio.CancelledError):
             await task
+        self.assertTrue(inner_task.done())
+        self.assertEqual(await inner_task, 'done')
         self.assertFalse(reached_end)
 
 
+class WaitForShieldTests(unittest.IsolatedAsyncioTestCase):
+
+    async def test_zero_timeout(self):
+        # With timeout=0 the task gets cancelled immediately
+        # and the shielded task is not run.
+        async def coro():
+            await asyncio.sleep(0.1)
+            return 'done'
+
+        task = asyncio.create_task(coro())
+        with self.assertRaises(asyncio.TimeoutError):
+            await asyncio.wait_for(asyncio.shield(task), timeout=0)
+
+        self.assertFalse(task.done())
+        self.assertFalse(task.cancelled())
```

Comment:
This seems to contradict the comment -- if it's not done, it will eventually run, right? So shield() is cancelled but the task is allowed to run in the background.

Is this the same for using timeout() directly?

Reply:
I have elaborated it more, should be easier to understand now. The task is scheduled to run in background, the updated test verifies it.